sw/device/silicon_creator/mask_rom/e2e_tests/mask_rom_e2e_testplan.hjson

```diff
@@ -0,0 +1,360 @@
+            - Issue RSTEN opcode.
+            - Issue RESET opcode.
+            - Verify that bootstrap was exited and that the chip was reset.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_phase2_page_program_op
+      desc: '''Verify that phase 2 of bootstrap correctly handles the PAGE_PROGRAM opcode.
+
+            - Enter bootstrap mode.
+            - Issue ERASE opcode to enter phase 2.
+            - Issue PAGE_PROGRAM opcode with an invalid address.
+            - Issue PAGE_PROGRAM opcode with a valid address.
+            - Issue RESET opcode to exit bootstrap and reset the chip.
+            - Verify that the valid address is correctly programmed and that the rest of flash is unchanged.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_phase2_erase_op
+      desc: '''Verify that phase 2 of bootstrap correctly handles the ERASE opcode.
+
+            - Enter bootstrap mode.
+            - Issue ERASE opcode to enter phase 2.
+            - Issue PAGE_PROGRAM opcode to program a target address.
+            - Issue ERASE opcode to the same target address.
+            - Issue RESET opcode to exit bootstrap and reset the chip.
+            - Verify that the target address was correctly erased and that the rest of flash is unchanged.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    // TODO: READ behavior is currently unspecified/not supported
+    {
+      name: mask_rom_e2e_phase2_read_op
+      desc: '''Verify that phase 2 of bootstrap correctly handles the READ opcode.
+
+            - Enter bootstrap mode.
+            - Issue ERASE opcode to enter phase 2.
+            - Issue PAGE_PROGRAM opcode to program a target address.
+            - Issue READ opcode to read the same target address.
+            - Verify that the READ behaves correctly.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+
+    // Chip-specific startup
+    {
+      name: mask_rom_e2e_startup_ast_init
+      desc: '''Verify that AST is initialized on correct lifecycle states.
+
+            - For each lifecycle state:
+              - Load an OTP image corresponding to this state
+              - Boot the chip
+              - Check the SENSOR_CTRL_STATUS_AST_INIT_DONE bit
+              - This bit should be 1 if the AST should be initialized and 0 otherwise.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_startup_main_sram_init
+      desc: '''Verify that the main SRAM is initialized correctly after POR.
+
+            - Initiate a POR.
+            - Wait until main SRAM is initialized.
+            - Check the following main SRAM registers in SRAM_CTRL.CTRL:
+              - SCR_KEY_VALID = 1
```

Comment:
since there will be an otp control over scrambling, will that be tested here?

Reply:
Added a TODO for this as noted by @timothytrippel.

Reply:
One more thing, could you raise this TODO to the top of the file, and file a corresponding issue to track?